### ISSUE TITLE
Next.js : Prop `id` did not match. Server: "pattern-61302" Client: "pattern-75795"

### ISSUE THREAD

**Issue by mfik** [wbkd/react-flow#2086]
### Describe the Bug

When trying basic integration within next.js, the following error returns. 

> Warning: Prop `id` did not match. Server: "pattern-61302" Client: "pattern-75795"
    at pattern
    at svg
    at Background (webpack-internal:///./node_modules/react-flow-renderer/dist/esm/index4.js:45:27)
    at Provider (webpack-internal:///./node_modules/zustand/esm/context.js:12:5)
    at Wrapper (webpack-internal:///./node_modules/react-flow-renderer/dist/esm/index.js:3701:23)
    at div
    at eval (webpack-internal:///./node_modules/react-flow-renderer/dist/esm/index.js:3749:20)
    at WorkflowConfigurator (webpack-internal:///./components/workflows/Configurator.tsx:68:62)
    at div
    at div
    at main
    at div
    at div
    at Layout (webpack-internal:///./components/layout/Layout.tsx:79:23)
    at AddCompanyPage
    at MaxFormsWeb (webpack-internal:///./pages/_app.tsx:44:27)
    at ErrorBoundary (webpack-internal:///./node_modules/next/dist/compiled/@next/react-dev-overlay/client.js:8:20584)
    at ReactDevOverlay (webpack-internal:///./node_modules/next/dist/compiled/@next/react-dev-overlay/client.js:8:23125)
    at Container (webpack-internal:///./node_modules/next/dist/client/index.js:357:9)
    at AppContainer (webpack-internal:///./node_modules/next/dist/client/index.js:791:26)
    at Root (webpack-internal:///./node_modules/next/dist/client/index.js:915:27) 

Code used is the basic example: 

`import { useState } from 'react';
import ReactFlow from 'react-flow-renderer';

const initialNodes = [
  {
    id: '1',
    type: 'input',
    data: { label: 'Input Node' },
    position: { x: 250, y: 25 },
  },

  {
    id: '2',
    // you can also pass a React component as a label
    data: { label: <div>Default Node</div> },
    position: { x: 100, y: 125 },
  },
  {
    id: '3',
    type: 'output',
    data: { label: 'Output Node' },
    position: { x: 250, y: 250 },
  },
];

const initialEdges = [
  { id: 'e1-2', source: '1', target: '2' },
  { id: 'e2-3', source: '2', target: '3', animated: true },
];

function Flow() {
  const [nodes, setNodes] = useState(initialNodes);
  const [edges, setEdges] = useState(initialEdges);

  return <ReactFlow nodes={nodes} edges={edges} fitView />;
}

export default Flow;`

### Steps to Reproduce the Bug or Issue

Install Next.js and react-flow and use the default example

### Expected behavior

To be able to integrate with next.js

### Platform

- OS: MaxOS
- Browser: Chrome
- Version: all

**Comment by bcakmakoglu** [comment]
To extend on the issue description:
The background generates a unique id on the server- and client-side which results in non-matching ids -> throws Error.

You can currently work around that though by making the Background a non-ssr component.

```jsx
import dynamic from "next/dynamic";

const NoSSRBackground = dynamic(
  () => import("react-flow-renderer").then((im) => im.Background),
  { ssr: false }
);
```
```jsx
<ReactFlow ...>
  <NonSSRBackground color="#aaa" gap={16} />
</ReactFlow>
```

Here's a [sandbox](https://codesandbox.io/s/focused-ellis-hv1svb?file=/src/Flow.js:144-180) with the NonSSRBackground.
You can force the error in the sandbox by just using the regular Background component.

**Comment by moklick** [comment]
thanks for bringing this up. we are no longer use math random for generating the ids. Fix released in v10.1.2
